types/target.go

```diff
@@ -43,6 +43,7 @@ type TargetConfig struct {
 	EventTags     map[string]string `mapstructure:"event-tags,omitempty" json:"event-tags,omitempty" yaml:"event-tags,omitempty"`
 	Gzip          *bool             `mapstructure:"gzip,omitempty" json:"gzip,omitempty" yaml:"gzip,omitempty"`
 	Token         *string           `mapstructure:"token,omitempty" json:"token,omitempty" yaml:"token,omitempty"`
+	SocksAddress  string            `mapstructure:"socks-address,omitempty" json:"socks-address,omitempty" yaml:"socks-address,omitempty"`
```

Comment:
Do you mind changing this field's name to just `proxy`? and derive the proxy type from the address scheme, e.g: `socks5://<address>`